scripts/swupd_performance/data_processing_script.py

```diff
@@ -0,0 +1,379 @@
+        logging.info("There are no files to be processed")
+        return
+
+    for fname in os.listdir(inputfolder):
+        logging.info("Processing File: %s", fname)
+        path = os.path.join(inputfolder, fname)
+
+        # if we fail to make sure file is json parsable
+        if not make_file_json_parsable(path):
+            mark_file_as_done(fname)
+            continue
+
+        # check if the output contains that system is up to update
+        if update_already_done(path):
+            mark_file_as_done(fname)
+            logging.info(
+                "file: %s says system is up-to-date", fname)
+            continue
+
+        if not skip_if_processed(fname):
+            start_processing_each_file(path, outfile)
+            mark_file_as_done(fname)
+        else:
+            logging.info("Skip processing file: %s", fname)
+
+
+def parse_timestamp_from_filename(fname):
+    """Parse & Extract timestamp from filename.
+
+    Parse each file_name to extract the
+    timestamp of its creation for the csv outfile field.
+
+    Args:
+        fname (string): name of file
+
+    Returns:
+        datatime obj or None if datetime parsing failed
+
+    """
+    date_obj = None
+    index = fname.find("file_")
+    try:
+        if index > -1:
+            timestamp = fname[index + 5:-5]
+            date_obj = datetime.strptime(timestamp, '%Y%m%d_%H%M%SZ')
+    except (IndexError, ValueError):
+        pass
+    return date_obj
+
+
+def start_processing_each_file(fname, outfile):
+    """Process file one by one.
+
+    Filter the json stream from each file
+    and the writes it to outfile. Called by
+    process.
+
+    Args:
+        fname (string): name of file
+        outfile (string): name of output csv file
+
+    """
+    if outfile is None:
+        outfile = "performance_sheet.csv"
+
+    outfile = os.path.abspath(outfile)
+
+    with open(fname) as infile:
+        data = json.load(infile)
+
+    # round 1 filtering
+    result = [x for x in data if x['type'] in FILTER1]
+
+    # round 2 filtering
+    result2 = []
+    for keyword in JSON_KEYWORD1:
+        temp = [x for x in result if x['msg'].find(keyword) > -1]
+        result2.extend(temp)
+
+    date_time = parse_timestamp_from_filename(fname)
+    if date_time:
+        result2.append(
+            {'msg': 'timestamp: ' + date_time.strftime('%Y%m%d_%H%M%S')})
+    else:
+        result2.append({'msg': 'timestamp: ' + "unknown"})
+    bundle_stats(result2, outfile)
+
+    # execution_time(result)
+
+
+def main():
+    """Enter through Main function."""
+    argparser = argparse.ArgumentParser()
+
+    argparser.add_argument("-f", "--folder", required=False,
+                           help="Folder to read results from, \
+                           will default to /var/log/swupd/staging")
+
+    argparser.add_argument(
+        "-o",
+        "--output",
+        required=False,
+        help="Abs path for CSV file to be created.. \
```

Comment:
..?
I think you have an extra "." and if so, default should be capitalized

Reply:
Done

Reply:
You mean the word default right

Reply:
Done